src/views/LotteryV2/components/BuyTicketsModal/TicketInput.tsx

```diff
@@ -0,0 +1,278 @@
+import React, { useState, useRef } from 'react'
+import styled from 'styled-components'
+import { Flex, Text } from '@pancakeswap/uikit'
+import { useTranslation } from 'contexts/Localization'
+import { Ticket, UpdateTicketAction } from './useTicketsReducer'
+
+const InputsContainer = styled.div<{ focused: boolean; isDuplicate: boolean }>`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 36px;
+  box-sizing: border-box;
+  border: 1px solid #d7caec;
+  background-color: #eeeaf4;
+  border-radius: 16px;
+  margin-bottom: 8px;
+  ${({ isDuplicate }) =>
+    isDuplicate &&
+    `
+    border: 1px solid #FFB237;
+    box-shadow: 0px 0px 0px 2px #FFB237;
+  `}
+  ${({ focused }) =>
+    focused &&
+    `
+    border: 1px solid #7645D9;
+    box-shadow: 0px 0px 0px 2px #E4DAF7;
+  `}
+`
+
+const DigitInput = styled.input`
+  border: none;
+  height: 32px;
+  padding: 0 12px;
+  font-size: 16px;
+  flex: 1;
+  width: 16px;
+  text-align: center;
+  min-width: 0;
+  background-color: transparent;
+  caret-color: #7a6faa;
+
+  &::placeholder {
+    text-align: center;
+  }
+
+  &:placeholder-shown {
+    text-align: left;
+  }
+
+  &:focus {
+    outline: none;
+  }
+
+  /* TODO check other browsers */
```

Comment:
Forgot to delete this comment. It works in FF, Chrome and Safari.
Not sure if webkit prefixes are needed though, but with stuff like spin-buttons and such I found that sometimes they don't get compiled correctly automatically, so maybe we should leave them in? Can test it later on different browsers without prefixes on prod built version.

Reply:
Yea I'd say leave them in for now Cheems. I'll add a note to the Trello refactors ticket to check & remove later